### PR TITLE
Update Solr container name, ckan-solr-dev is obsolete

### LINKED ISSUE #99: Github actions workflows fail on docker container pull
```
Starting solr service container
  /usr/bin/docker pull ckan/ckan-solr-dev:2.9
  Error response from daemon: pull access denied for ckan/ckan-solr-dev, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  Warning: Docker pull failed with exit code 1, back off 7.312 seconds before retry.
  /usr/bin/docker pull ckan/ckan-solr-dev:2.9
  Error response from daemon: pull access denied for ckan/ckan-solr-dev, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  Warning: Docker pull failed with exit code 1, back off 9.989 seconds before retry.
  /usr/bin/docker pull ckan/ckan-solr-dev:2.9
  Error response from daemon: pull access denied for ckan/ckan-solr-dev, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
  Error: Docker pull failed with exit code 1
```

### PR DESCRIPTION
**Pull request by ThrawnCA**
Fixes #99

### CODE REVIEW

**Comment by wardi** [comment]
@ThrawnCA broken test is unrelated?

**Comment by ThrawnCA** [comment]
> @ThrawnCA broken test is unrelated?

It looks unrelated, yes. This PR only changes the Solr container.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
       image: openknowledge/ckan-dev:${{ matrix.ckan-version }}
     services:
       solr:
-        image: ckan/ckan-solr-dev:${{ matrix.ckan-version }}
+        image: ckan/ckan-solr:${{ matrix.ckan-version }}
```
Comment (amercader) [inline .github/workflows/test.yml:29]:
```suggestion
        image: ckan/ckan-solr:${{ matrix.ckan-version }}-solr9
```

Comment (amercader) [reply .github/workflows/test.yml:29]: We are using this variant to future proof the image name. Other than that looks good, feel free to merge

**Pull request merged**